chevah/keycert/ssh.py

```diff
@@ -629,6 +629,13 @@ def _fromString_BLOB(cls, blob):
         @raises BadKeyError: if the key type (the first string) is unknown.
         """
         keyType, rest = common.getNS(blob)
+
+        try:
+            # keyType is supposed to contain only ascii characters
+            keyType.decode('ascii')
+        except UnicodeDecodeError:
+            raise BadKeyError('Invalid non-ascii blob type.')
```

Comment:
For such errors please also include the truncated repr of 'keyType' in the error message.

It will not help the end user, but it will help the development/support team when an user is reporting an error.

The user report an error like `Invalid non-ascii blob type.` but then we will have no idea if this is a bug in our software or user provided invalid input.

If user will report something like `Invalid non-ascii blob type. bla-bla` we can get an idea of what was going on there.

---------

I am not sure if this is required. I prefer to leave this out and have this case handled by `BadKeyError('Unknown blob type: %r' % keyType[:30])` 

As a general design rule, we should just use EAFF and make sure the error messages are sane.

------

While keyType are expected to be ascii, in the RFC i have not found any restriction for that... so they should be treated as bytes.

We should update chevah-keycert and port it to Py3 (already done in Twisted) and after that it will be clear which value is text and which is bytes.